src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocations.Fixer.cs

```diff
@@ -0,0 +1,199 @@
+
+        public sealed override FixAllProvider GetFixAllProvider() =>
+            WellKnownFixAllProviders.BatchFixer;
+
+        public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+        {
+            Document doc = context.Document;
+            CancellationToken ct = context.CancellationToken;
+            SyntaxNode root = await doc.GetSyntaxRootAsync(ct).ConfigureAwait(false);
+
+            if (!(root.FindNode(context.Span, getInnermostNodeForTie: true) is SyntaxNode node))
+            {
+                return;
+            }
+
+            SemanticModel model = await doc.GetSemanticModelAsync(ct).ConfigureAwait(false);
+
+            // The analyzer created the diagnostic on the IdentifierNameSyntax, and the parent is the actual invocation
+            if (!TryGetInvocation(model, node, ct, out IInvocationOperation? invocation))
+            {
+                return;
+            }
+
+            if (!TryGetAncestorDeclarationCancellationTokenParameterName(node, out string? parameterName))
+            {
+                return;
+            }
+
+            string title = MicrosoftNetCoreAnalyzersResources.ForwardCancellationTokenToInvocationsTitle;
+
+            if (!TryGenerateNewDocumentRoot(doc, root, invocation, parameterName, out SyntaxNode? newRoot))
+            {
+                return;
+            }
+
+            Task<Document> createChangedDocument(CancellationToken _) =>
+                Task.FromResult(doc.WithSyntaxRoot(newRoot));
+
+            context.RegisterCodeFix(
+                new MyCodeAction(
+                    title: title,
+                    createChangedDocument,
+                    equivalenceKey: title),
+                context.Diagnostics);
+        }
+
+        private bool TryGenerateNewDocumentRoot(
+            Document doc,
+            SyntaxNode root,
+            IInvocationOperation invocation,
+            string cancellationTokenParameterName,
+            [NotNullWhen(returnValue: true)] out SyntaxNode? newRoot)
+        {
+            newRoot = null;
+
+            SyntaxGenerator generator = SyntaxGenerator.GetGenerator(doc);
+
+            // Pass the same arguments and add the ct
+            List<SyntaxNode> newArguments = new List<SyntaxNode>();
+            bool shouldTokenUseName = false;
+            string paramName = string.Empty;
+
+            // In C#, invocation.Arguments contains the arguments in the order passed by the user
+            // In VB, invocation.Arguments contains the arguments in the official parameter order
+            for (int i = 0; i < invocation.Arguments.Length; i++)
+            {
+                IArgumentOperation argument = invocation.Arguments[i];
+
+                // The type name is detected even if using an alias
+                if (!argument.Parameter.Type.Name.Equals("CancellationToken", StringComparison.Ordinal))
```

Comment:
[suggestion] If you revert the condition and add a `continue;` you can reduce quite much the nesting.

Reply:
The code changed considerably so this does not apply anymore.